sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQueryHashPartitionVerifySuite.scala

```diff
@@ -0,0 +1,219 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.streaming
+
+import java.io.{BufferedWriter, DataInputStream, DataOutputStream, File, FileInputStream, FileOutputStream, FileWriter}
+
+import scala.io.Source
+import scala.util.Random
+
+import com.google.common.io.ByteStreams
+
+import org.apache.spark.SparkConf
+import org.apache.spark.io.CompressionCodec
+import org.apache.spark.sql.{RandomDataGenerator, Row}
+import org.apache.spark.sql.catalyst.CatalystTypeConverters
+import org.apache.spark.sql.catalyst.expressions.{BoundReference, GenericInternalRow, UnsafeProjection, UnsafeRow}
+import org.apache.spark.sql.catalyst.plans.physical._
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.types.{BinaryType, DataType, DoubleType, FloatType, IntegerType, StringType, StructType}
+
+class StreamingQueryHashPartitionVerifySuite extends StreamTest {
+  override protected def sparkConf: SparkConf =
+    super.sparkConf
+      .set(SQLConf.SHUFFLE_PARTITIONS.key, numShufflePartitions.toString)
+
+  // Configs for golden file
+  private val goldenFileURI =
+    this.getClass.getResource("/structured-streaming/partition-tests/").toURI
+
+  private val schemaFileName = "randomSchemas" // files for storing random input schemas
+  private val rowAndPartIdFilename =
+    "rowsAndPartIds" // files for storing random input rows and resulting partition ids
+  private val codec = CompressionCodec.createCodec(
+    sparkConf) // Used for compressing output to rowAndPartId file
+
+  // Configs for random schema generation
+  private val variableTypes = Seq(IntegerType, DoubleType, FloatType, BinaryType, StringType)
```

Comment:
While I don't think we have to add all types (size restriction), let's add importantly used types. StructType with two TimestampTypes consists a type of time window, so `TimestampType` would be a good one to test. `LongType` is also a good one.

Reply:
Added.